src/library_manager/llext_manager.c

```diff
@@ -129,6 +152,39 @@ static int llext_manager_load_data_from_storage(const struct llext_loader *ldr,
 	return ret;
 }
 
+static void llext_manager_unmap_detached_sections(const struct llext_loader *ldr,
+						  const struct llext *ext,
+						  enum llext_mem region,
+						  void __sparse_cache *vma,
+						  size_t size)
+{
+#ifdef CONFIG_MMU
+	unsigned int i;
+	const void *region_addr;
+
+	llext_get_region_info(ldr, ext, region, NULL, &region_addr, NULL);
+
+	for (i = 0; i < llext_section_count(ext); i++) {
+		const elf_shdr_t *shdr;
+		enum llext_mem s_region = LLEXT_MEM_COUNT;
+		size_t s_offset = 0;
+
+		llext_get_section_info(ldr, ext, i, &shdr, &s_region, &s_offset);
+
+		/* skip sections not in the requested region */
+		if (s_region != region)
+			continue;
+
+		/* unmap detached sections (will be outside requested VMA area) */
```

Comment:
if you have to update this PR, better replace "unmap" with "restore default flags"

Reply:
Lets do this incrementally - CI passing

Reply:
It was unstable with many timeouts in different tests each run, with reverted 'ipc: move most functions to run from DRAM' its now stable in linux CI, lets firstly merge base PTL DRAM enablement and later stabilize IPC

Reply:
@abonislawski ok, so you're referring to jenkins failures, not to your local or "Intel Internal" tests. One of the failing jenkins runs was https://sof-ci.01.org/sofpr/PR9893/build11697/devicetest/index.html. In it we see the firmware running into an error:
```
[00:00:02.029,105] <err> ipc: ipc_comp_free: ipc_comp_free(): comp id: 0x3 state is 5 cannot be freed
[00:00:02.029,115] <err> ipc: ipc_pipeline_free: ipc_pipeline_free(): module free () failed
[00:00:02.029,128] <err> ipc: ipc_cmd: ipc4: FW_GEN_MSG failed with err 12
```
So that should be debuggable?

Reply:
Yes, only linux CI, not a problem for another one in both configurations. This specific fail with pass in most runs and I really dont want to debug all of this, we need DRAM for modules on platforms which DRAM functionality

Reply:
Actually I think reverting the DRAM IPC commit only hides the problem. I'd propose a test: update to the Zephyr version as in the PR (use the first commit as is), disable LLEXT and enable cold storage on PTL and run that through the CI. I expect that to pass. Then enable modules but remove all `__cold` attributes from them - that should pass too. So I suspect only when you enable `__cold` code in LLEXT modules, then your flag updating code breaks it. So maybe we still have some page overlaps? You're updating flags for sections, that shouldn't be touched?

Reply:
unfortunately this scenario with __cold removed in src/audio also failed